tests/codex/merkletree/testmerkledigest.nim

```diff
@@ -0,0 +1,56 @@
+import std/unittest
+import std/sequtils
+import std/random
+
+import pkg/constantine/math/arithmetic
+
+import pkg/poseidon2
+import pkg/poseidon2/io
+import pkg/poseidon2/sponge
+
+import pkg/questionable/results
+
+import pkg/codex/merkletree
+import pkg/codex/utils/digest
+
+suite "Digest - MerkleTree":
+
+  const KB = 1024
+
+  test "Hashes chunks of data with sponge, and combines them in merkle root":
+    let bytes = newSeqWith(64*KB, rand(byte))
+    var leaves: seq[Poseidon2Hash]
+    for i in 0..<32:
+      let
+        chunk = bytes[(i*2*KB)..<((i+1)*2*KB)]
+        digest = Sponge.digest(chunk, rate = 2)
+      leaves.add(digest)
+
+    let
+      digestTree = Poseidon2MerkleTree.digest(bytes, chunkSize = 2*KB).tryGet
+      tree = Poseidon2MerkleTree.init(leaves).tryGet
+      root = tree.root.tryGet
+      expected = digestTree.root.tryGet
```

Comment:
You could test against the implementation in nim-poseidon2:
```suggestion
      expected = SpongeMerkle.digest(bytes, chunkSize = 2*KB)
```
This makes the test simpler, and stronger because you're testing against a different implementation.

Reply:
We should probably have both checks